### PR TITLE
chore(query): add full expr in error message

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

add full expr in error message
## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16032)

### CODE REVIEW

File: src/query/expression/src/evaluator.rs [hunk 1]
```diff
@@ -125,6 +125,18 @@ impl<'a> Evaluator<'a> {
 
     pub fn run(&self, expr: &Expr) -> Result<Value<AnyType>> {
         self.partial_run(expr, None, &mut EvaluateOptions::default())
+            .map_err(|err| {
+                if err
+                    .message()
+                    .contains(format!("`{}`", expr.sql_display()).as_str())
```
Comment (zhang2014) [inline src/query/expression/src/evaluator.rs:131]: Please add large expr test to avoid stack overflow

File: src/query/functions/tests/it/scalars/testdata/array.txt [hunk 2]
```diff
@@ -11,7 +11,7 @@ error:
   --> SQL:1:2
   |
 1 | ['a', 1]
-  |  ^^^ invalid digit found in string while evaluating function `to_uint64('a')` in expr `to_uint64('a')`
+  |  ^^^ invalid digit found in string while evaluating function `to_uint64('a')` in expr `to_uint64('a')`, during run expr: `array(to_uint64('a'), to_uint64(1))`
```
Comment (b41sh) [inline src/query/functions/tests/it/scalars/testdata/array.txt:14]: `in expr` seems duplicate with `during run expr`, may be we can remove `in expr` here?

Comment (sundy-li) [reply src/query/functions/tests/it/scalars/testdata/array.txt:14]: It's not duplicated, this case is using literal in expr.


see:

```
  |              ^ divided by zero while evaluating function `divide(1, 0)` in expr `1 / expr_a`, during run expr: `if(CAST(cond_a AS Boolean NULL), 1 / expr_a, to_float64(expr_else))`

```
